### DISCUSSION TITLE
Remote submit

### DISCUSSION THREAD

**Discussion by Sxoo** [react-hook-form/react-hook-form#2024] [Q&A] [answered]
Hi,

My current app flow is 1st form -> submit 1st form -> 2nd form -> submit 2nd -> 3rd -> submit 3rd.
It works fine if the submit buttons are inside the form. I would like to upgrade my flow with adding a Stepper into my Header component so that I could submit 3 different forms from inside the Header. How should I do that? 

Currently struggling to reach the handleSubmit function inside my form from the Header component. And afterwards, how do I make the 3 different forms distinct should I name them and pass a name with data to onSubmit, or just create different handleSubmit functions? 

The example I followed - https://react-hook-form.com/advanced-usage#ConnectForm

Edit: vague code sandbox representation
https://codesandbox.io/s/great-matsumoto-ukm99?fontsize=14&hidenavigation=1&theme=dark

**Comment by bluebill1049** [comment, marked as answer]
`handleSubmit()()` is what you after

**Reply by Sxoo** [reply]
after handleSubmit()() ----> Unhandled Rejection (TypeError): callback is not a function

**Reply by bluebill1049** [reply]
can you update your codesandbox?

**Reply by Sxoo** [reply]
I believe that I did https://codesandbox.io/s/great-matsumoto-ukm99?fontsize=14&hidenavigation=1&theme=dark&file=/src/Header.js

**Reply by bluebill1049** [reply]
i don't see the error, what's the steps to reproduce?

**Reply by Sxoo** [reply]
Clicking "This doesn't" should log the form data, which it doesn't.

**Reply by bluebill1049** [reply]
u miss the call back inside: https://codesandbox.io/s/remote-submit-68ysx

**Reply by Sxoo** [reply]
could you provide me with a more in depth example, I'm not quite sure how to pull this off

**Reply by bluebill1049** [reply]
`handleSubmit` need a callback as simple as that.

**Reply by Sxoo** [reply]
onSubmit={methods.handleSubmit(onFormSubmit, ( ) => {
          console.log("Do I understand you correctly?");
        })}

**Reply by bluebill1049** [reply]
```
onSubmit={methods.handleSubmit(( ) => {
  console.log("Do I understand you correctly?");
})}
```

**Reply by Sxoo** [reply]
methods.handleSubmit()();  <--- still callback is not a function in Header.js

**Reply by bluebill1049** [reply]
methods.handleSubmit(() => console.log('please give a callback here'))();

**Reply by Sxoo** [reply]
alright I got it and lastly how do I pass my form data to Header.js callback?

**Reply by mortmoe** [reply]
You would need to put both your form with nested stuff and your header inside the same <FormProvider> don't you? 
I would do something like this: 
- In my main app (MainInfo in your case) I'd get methods from useForm, create a _handleSubmit that you can pass on using methods.handleSubmit(data => .... do something with data)
- Wrap both the form, header and whatever else you need to interact with the form inside <Formprovider {...methods}>
- Pass your _handleSubmit to the component that need it
Something like this: 
https://codesandbox.io/s/great-matsumoto-ukm99?fontsize=14&hidenavigation=1&theme=dark&file=/src/App.js

**Reply by Sxoo** [reply]
Yes that would be a solution @mortmoe but I am trying to find a way around this because in the end I will have a single header which navigates 3 different forms. I feel like your solution would leave me with a single header and form or with 3 headers and 3 forms. 

But thanks @mortmoe @bluebill1049  for your answers I might just resort to a simpler structure and then use the answers you both provided me. :)

**Reply by mortmoe** [reply]
No problem. But you would still only need one header... just change the _handleSubmit if you load another form :)

**Reply by tjdam** [reply]
Is there a way to use the handleSubmit function already associated with the form? Or the only way would be to pass the same callback to handleSubmit()()?
 
I ask because I'm trying to call handleSubmit from a child component using FormContext
